addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5464,6 +5468,20 @@ registry.layout_column = SnippetOptionWidget.extend(ColumnLayoutMixin, {
         // Add the new column and update the grid height.
         rowEl.appendChild(newColumnEl);
         gridUtils._resizeGrid(rowEl);
+
+        // Scroll to the new column if more than half of it is hidden (= out of
+        // the viewport or hidden by an other element).
+        if (elementType === "image") {
+            // If an image was added, wait for it to be loaded before scrolling.
+            await imageLoadedPromise;
+        }
```

Comment:
I did a quick check and all images I add have the same size. Do we really need to wait for image to load to caclulate its middle point ? If we know its size in advance we could scroll before the image is loaded no ?

Reply:
If I don't do that, the screen doesn't scroll even though it should, since it is supposed to only look at the column position -> it does execute the code but the scroll doesn't happen visually on the page

At first, I thought it was because maybe the media dialog was not closed yet and was blocking the scroll (so a race condition), but when testing, `scrollIntoView` still works with the modal open so it is not that

However, I noticed that sometimes it did scroll (when uploading smaller images or choosing an already uploaded one), so I thought that it had something to do with the image loading, which is why I await it
=> I have to wait for something and it seemed to be the most logical one